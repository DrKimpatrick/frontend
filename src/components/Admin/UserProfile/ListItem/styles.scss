.listItems {
  position: relative;
  color: #3b3a3a;
  li {
    padding: 5px;
    border-radius: 2px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding-left: 20px;
    padding-right: 20px;
    h5 {
      font-size: 15px;
    }
    .selectItems {
      position: relative;
      width: 150px;
      border: none;
      border-radius: 5px;
      background: none;
      select {
        color: #727272;
        width: 100%;
        border: none;
        padding: 8px;
        background: none;
        appearance: none;
        font-size: 15px;
      }
      .customArrow {
        position: absolute;
        top: 0;
        right: 0;
        height: 100%;
        background-color: none;
        width: 2rem;
        pointer-events: none;
        display: flex;
        justify-content: center;
        align-items: center;
      }
    }
  }
}
