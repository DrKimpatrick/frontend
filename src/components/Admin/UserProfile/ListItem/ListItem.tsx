import React from 'react';
import './styles.scss';
import { ArrowDropDown } from '@material-ui/icons';

export interface ListItemProps {
  id: string;
  name: string;
  status?: string;
}

interface Props {
  listItem: ListItemProps[];
}

const options = [
  { label: 'Verified', value: 'Verified' },
  { label: 'Not Verified', value: 'Not Verified' }
];

const ListItem = (props: Props) => {
  const { listItem } = props;

  return (
    <div className="listItems">
      <ul className="p-0 m-2 bg-card-preview">
        {listItem.length > 0 &&
          listItem.map((item, index) => (
            <li className="py-2 py-2 rounded-sm" key={index}>
              <h5>{item.name}</h5>
              <div className="selectItems">
                <select
                  value={item.status ? item.status : ''}
                  onChange={e => e.target.value}
                >
                  {options.map((opt, i) => (
                    <option value={opt.value} key={i}>
                      {opt.label}
                    </option>
                  ))}
                </select>
                <span className="customArrow">
                  <ArrowDropDown />
                </span>
              </div>
            </li>
          ))}
      </ul>
    </div>
  );
};

export default ListItem;
