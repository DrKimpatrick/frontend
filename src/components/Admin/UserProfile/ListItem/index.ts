import ListItem from './ListItem';

export { ListItem };
export * from './ListItem';
