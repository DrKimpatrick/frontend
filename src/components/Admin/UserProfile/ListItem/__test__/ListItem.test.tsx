import React from 'react';
import ReactDom from 'react-dom';
import ListItem from '../ListItem';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

const props = {
  listItem: {
    id: 'id',
    name: 'name',
    status: 'status'
  }
};

describe('List Item', () => {
  afterEach(cleanup);

  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDom.render(<ListItem {...props} />, div);
  });

  it('should create snapshot', () => {
    const tree = renderer.create(<ListItem {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
