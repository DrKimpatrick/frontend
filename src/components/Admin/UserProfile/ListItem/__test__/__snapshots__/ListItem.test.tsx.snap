// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item should create snapshot 1`] = `
<div
  className="listItems"
>
  <ul
    className="p-0 m-2 bg-card-preview"
  >
    <li
      className="py-2 py-2 rounded-sm"
    >
      <h5>
        name
      </h5>
      <div
        className="selectItems"
      >
        <select
          onChange={[Function]}
          value="status"
        >
          <option
            value="Verified"
          >
            Verified
          </option>
          <option
            value="Not Verified"
          >
            Not Verified
          </option>
        </select>
        <span
          className="customArrow"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </span>
      </div>
    </li>
  </ul>
</div>
`;
